From 85080217468bfde95e235bd4498b44e5617e6510 Mon Sep 17 00:00:00 2001
From: xufuji456 <839789740@qq.com>
Date: Sun, 5 Mar 2023 21:59:45 +0800
Subject: [PATCH] codec/arm/hevcdsp_idct_neon: remove duplicate mov

---
 libavcodec/arm/hevcdsp_idct_neon.S | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/libavcodec/arm/hevcdsp_idct_neon.S b/libavcodec/arm/hevcdsp_idct_neon.S
index 41ca3b83a8..66ed1c6785 100644
--- a/libavcodec/arm/hevcdsp_idct_neon.S
+++ b/libavcodec/arm/hevcdsp_idct_neon.S
@@ -877,33 +877,27 @@ function func_tr_32x4_\name
         vld1.s16        {q0}, [r9, :128]!
         vld1.s16        {q1}, [r9, :128]
         add             r4, sp, #2048
+        mov             r2, #64
+        mov             r8, #-64
 
         bl              tr_block1
         mov             r1, r11
-        mov             r2, #64
-        mov             r8, #-64
         add             r3, r11, #(56 + 3 * 64)
         scale_store     \shift
 
         bl              tr_block2
         add             r1, r11, #8
         add             r3, r11, #(48 + 3 * 64)
-        mov             r2, #64
-        mov             r8, #-64
         scale_store     \shift
 
         bl              tr_block3
         add             r1, r11, #16
         add             r3, r11, #(40 + 3 * 64)
-        mov             r2, #64
-        mov             r8, #-64
         scale_store     \shift
 
         bl              tr_block4
         add             r1, r11, #24
         add             r3, r11, #(32 + 3 * 64)
-        mov             r2, #64
-        mov             r8, #-64
         scale_store     \shift
 
         bx               r10
-- 
2.32.0 (Apple Git-132)

